builders/xcm/xcm-transactor.md

```diff
@@ -0,0 +1,194 @@
+ - **location** — a multilocation representing a chain in the ecosystem. The value is used to set or retrieve the transact information
+ - **destination** — a multilocation representing a chain in the ecosystem where the XCM message is being sent to
+ - **currencyID** — the ID of the currency being used to pay for the remote call execution. Different runtimes have different ways of defining the IDs. In the case of Moonbeam-based networks, `SelfReserve` refers to the native token, and `ForeignAsset` refers to the asset ID of the XC-20 (not to be confused with the XC-20 address)
+ - **destWeight** — the maximum amount of execution time you want to provide in the destination chain to execute the XCM message being sent. If not enough weight is provided, the execution of the XCM will fail, and funds might get locked in either the sovereign account or a special pallet. For transacts through derivative, you have to take into account the `asDerivative` extrinsic as well, but the XCM-Transactor pallet adds the weight for the XCM instructions with the transact information set before. **It is essential to correctly set the destination weight to avoid failed XCM executions**
+ - **innerCall** — encoded call data of the call that will be executed in the destination chain. This is wrapped with the `asDerivative` option if transacting through the derivative account
+ - **feeLocation** — a multilocation representing the currency being used to pay for the remote call execution
+ - **feePayer** — the address that will pay for the remote XCM execution in the transact through sovereign extrinsic. The fee is charged in the corresponding [XC-20 token](/builders/xcm/xc20/overview/)
+ - **call** — similar to `innerCall`, but it is not wrapped with the `asDerivative` extrinsic
+ - **originKind** — dispatcher of the remote call in the destination chain. There are [four types of dispatchers](https://github.com/paritytech/polkadot/blob/0a34022e31c85001f871bb4067b7d5f5cab91207/xcm/src/v0/mod.rs#L60){target=_blank} available
+
+The XCM-Transactor pallet provides three read-methods:
+
+ - **indexToAccount(index)** — returns the origin chain account associated with the given derivative index
+ - **transactInfoWithWeightLimit(location)** — returns the transact information for a given multilocation
+ - **palletVersion()** — returns current pallet version from storage
+
+## Building an XCM with the XCM-Transactor Pallet {: #build-xcm-xcmtransactor-pallet}
+
+This guide covers building an XCM message for remote executions using the XCM-Transactor pallet, specifically with the `transactThroughDerivative` function. The steps to use the `transactThroughDerivativeMultilocation` function are identical, but instead of the currency ID, you specify a multilocation for the fee token.
+
+!!! note
+    You need to ensure that the call you are going to execute remotely is allowed in the destination chain!
```

Comment:
I think it would be good to know here if the allowed remote calls are on a whitelist, but generally not allowed, or generally all calls are allowed, but certain specific ones are blocked. Or if this varies by parachain, and in that case, if Moonbeam allows remote call or disallows them by default.

Reply:
This is fairly complex. I guess because it is a parachain by parachain scenario, the developer would need to ask first, or try it out on a TestNet.